src/lib/browse/layers/url.ts

```diff
@@ -0,0 +1,45 @@
+import { writable, type Writable } from "svelte/store";
+
+// Create a store to represent whether a layer should be shown or hidden. The
+// state is synced as a URL query parameter.
+export function showHideLayer(name: string): Writable<boolean> {
+  let initialValue = new URLSearchParams(window.location.search).has(name);
+  let store = writable(initialValue);
+  // TODO How do we avoid leaking this?
+  store.subscribe((show) => {
+    let url = new URL(window.location.href);
+    if (show) {
+      url.searchParams.set(name, "1");
+    } else {
+      url.searchParams.delete(name);
+    }
+    window.history.replaceState(null, "", url.toString());
+  });
+  return store;
+}
+
+// Create a store to represent some state about a layer, syncing it to a URL
+// query parameter. The parameter missing is equivalent to stringify returning
+// null.
+export function customUrlState<T>(
+  name: string,
+  defaultValue: T,
+  stringify: (state: T) => string | null,
+  parse: (param: string) => T,
+): Writable<T> {
+  let param = new URLSearchParams(window.location.search).get(name);
+  let initialValue = param == null ? defaultValue : parse(param);
+  let store = writable(initialValue);
+  // TODO How do we avoid leaking this?
```

Comment:
In prod, this won't be a problem -- each layer component is created only once, so there'll only be one store for the lifetime of the app. In dev mode with HMR, this will leak memory, because I couldn't figure out a clean way to call unsubscribe / teardown the stores. We could add onDestroy handlers to every single component I guess, but seems like overkill for a dev-mode mostly-invisible problem